### PR TITLE
364 refactor featured products pages

### LINKED ISSUE #466: add support for featured
add a menu in the topbar with a dropdown with featured products. These can be different depending on the cluster. #364 also relevant for this

### LINKED ISSUE #478: move default values to a single location
We should move all the default values that we use in a single location so it's easier to maintain

https://github.com/vyper-protocol/vyper-otc-ui/blob/5a7a9a1fce56fedcbf007d74354c350de438376d/src/pages/contract/create/index.tsx#L43-L63

### LINKED ISSUE #495: Include support for black scholes pricer
Some examples:
https://github.com/bret-blackford/Black-Scholes-in-JavaScript
(with explanation) https://www.grapecity.com/blogs/how-to-create-javascript-option-pricing-calculator-black-scholes-model

### PR DESCRIPTION
**Pull request by vantessel**
fix #495 
fix #478 
fix #466

### CODE REVIEW

File: src/pages/contract/featured/sol-10x/index.tsx [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+				if (!showRfq) {
+					setShowRfq(true);
+				}
+				setRfqProgress(100);
+				setRfqLoading(false);
+			}
+		}, 1500);
+
+		return () => clearTimeout(timer);
+	}, [showRfq, rfqLoading, pricesValue]);
```
Comment (thegreat-vanderlinde) [inline src/pages/contract/featured/sol-10x/index.tsx:70]: would change this logic in a more readble version, like:
```ts
const onRfqButtonClick = async () => {
		setRfqLoading(true);

		try {
			await sleep(1500);

			setSettleStart(refSettle());
			setRfqStrike(BSDigitalStrike(pricesValue[0], 0, 3.5, 1 / 24 / 365, true, 0.1));
			if (!showRfq) {
				setShowRfq(true);
			}
		} finally {
			setRfqProgress(100);
			setRfqLoading(false);
		}
	};
```

`sleep` is from `utils/sleep`

File: src/pages/contract/featured/sol-10x/index.tsx [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+				// 60s timer updated every 0.5s
+				return Math.max(0, oldProgress - 100 / 120);
+			});
+		}, 500);
+
+		return () => {
+			clearInterval(timer);
+		};
+	}, [rfqProgress]);
```
Comment (thegreat-vanderlinde) [inline src/pages/contract/featured/sol-10x/index.tsx:87]: using `setInterval` in a `useEffect` is usually dangerous. Having the `rfqProgress` as a dependency and changing it in the timer body will create a new timer on each tick (you can try to console log the timer id on line 83 to notice the new id every time). 

A better solution could be to start the timer at the page mount (empty array as `useEffect` dependency) and simply decreasing the progress value everytime:
```ts
useEffect(() => {
		const timer = setInterval(() => {
			setRfqProgress((oldProgress) => {
				return Math.max(0, oldProgress - 100 / 120);
			});
		}, 500);
		return () => clearInterval(timer);
	}, []);
```

this should work fine, having the rfqProgress reset to 100 on the button click

**Review by thegreat-vanderlinde — CHANGES_REQUESTED** [review]
not sure if it has been introduced with this PR but we have a lot of errors in the console:
Screenshot 2022-12-19 at 16 52 31

they seems triggered on a regular schedule.

we should investigate if we're throwing them in a timer or they're triggered by trading view component (in that case we could eventually investigate if it's possible to suppress)

**Comment by vantessel** [comment]
The errors seem to be triggered by the loading of an external script from [tradingview](https://www.tradingview.com/widget/advanced-chart/). I tried embedding directly but I see the same errors. I didn't find an easy way to suppress them, any suggestion?

Same error [here](https://k-128.github.io/react-tradingview-embed/?path=/story/widgets--advanced-chart-story)

**Comment by thegreat-vanderlinde** [comment]
> The errors seem to be triggered by the loading of an external script from [tradingview](https://www.tradingview.com/widget/advanced-chart/). I tried embedding directly but I see the same errors. I didn't find an easy way to suppress them, any suggestion?
> 
> Same error [here](https://k-128.github.io/react-tradingview-embed/?path=/story/widgets--advanced-chart-story)

I tried to investigate a little, but haven't found anything viable. for now we can ignore and merge as is

**Pull request merged**